### DISCUSSION TITLE
What's the font of Termux? Is there a .ttf?

### DISCUSSION THREAD

**Discussion by 1457384613gh** [termux/termux-app#3529] [Q&A] [answered]
What's the font of Termux? 
Is there a .ttf?

**Comment by agnostic-apollo** [comment, marked as answer]
https://wiki.termux.com/wiki/Termux:Styling

**Reply by 1457384613gh** [reply]
Thank you for your reply.
But I want to get the original font of termux.


Look at this picture, which font do the white words use?

![Screenshot_2023-07-27-10-03-35-627_com.termux-edit.jpg](https://github.com/termux/termux-app/assets/69227436/c433e7d8-c0bc-4452-ba15-6800b579c5ee)

**Reply by agnostic-apollo** [reply]
That's android system's default monospace font depending on device and android version.

Check `/system/etc/fonts.xml` file for system fonts config. Font ttf files are under `/system/fonts`.

To revert to default font in termux, delete the custom ttf file `~/.termux/font ttf`. Color can be reverted by deleting `~/.termux/colors.properties` file.

https://www.xda-developers.com/roboto-android-fall-rise-default-font

**Reply by 1457384613gh** [reply]
Look at this picture, this is not the default font of android system.

The font of buttons below is the default font of android system.

But the font of terminal is …


![Screenshot_2023-07-27-12-23-51-583_com.termux-edit.jpg](https://github.com/termux/termux-app/assets/69227436/e6e62f42-78e4-4eaf-9df0-1409fbb853e1)

**Reply by agnostic-apollo** [reply]
Extra keys do not use monospace font, they use default normal fonts. Android uses `DroidSansMono` as monospace and `Roboto` as normal font by default.

https://cs.android.com/android/platform/superproject/+/android-13.0.0_r54:frameworks/base/data/fonts/fonts.xml;l=25

https://cs.android.com/android/platform/superproject/+/android-13.0.0_r54:frameworks/base/data/fonts/fonts.xml;l=237

**Comment by dzabel-de** [comment]
Maybe this helps somebody:
I have been using Termux since I own this phone (motorola edge 30 lite), which came with Android 11. Uprdade to Android 12 worked more or less well (afair there were some problems due to stricter file system permissions, not quite sure, though). But after upgrade to Android 13, termux had a real ugly font, the lower case letters i and l showing up very thick. Ther was no file ~/.termux.font.ttf.
I did a "cp /system/fonts/DroidSansMono.ttf ~/.termux/font.ttf" and now it looks fine again!

**Reply by n9g** [reply]
Thanks for your post!
